content/influxdb/v1.8/concepts/tsi-details.md

```diff
@@ -86,12 +85,29 @@ The following occurs when a write comes into the system:
 
 ### Compaction
 
-Once the LogFile exceeds a threshold (5MB), then a new active log file is created and the previous one begins compacting into an IndexFile.
-This first index file is at level 1 (L1).
-The log file is considered level 0 (L0).
+When compactions are enabled, every second, InfluxDB checks to see whether compactions are needed. If there haven't been writes during the `compact-full-write-cold-duration` period (by default, `4h`), InfluxDB compacts all TSM files. Otherwise, InfluxDB groups TSM files into compaction levels (determined by the number of times the file have been compacted), and attempts to combine files and compress them more efficiently.
+
+Once the `LogFile` exceeds a threshold (`5MB`), InfluxDB creates a new active log file, and the previous one begins compacting into an `IndexFile`. This first index file is at level 1 (L1). The log file is considered level 0 (L0). Index files can also be created by merging two smaller index files together.
+For example, if two contiguous L1 index files exist, InfluxDB merges them into an L2 index file.
+
+InfluxDB schedules compactions preferentially, using the following guidelines:
+
+- The lower the level (the fewer times a file has been compacted), the more weight is given to compacting it.
+- The more compactible files in a level, the higher the priority given to that level. If the number of files in each level is equal, lower levels are compacted first.
+- If a higher level has more candidates for compaction, it may be compacted before a lower level. The equation is the number of candidate files for compaction times level weights of 0.4, 0.3, 0.2, and 0.1.
```

Comment:
@davidby-influx the equation to calculate the compaction priority?....
...the weight is multiplied by the number of compactible files in any given level --no matter the level?